node/lib/util/rebase_util.js

```diff
@@ -120,193 +108,106 @@ const callFinish = co.wrap(function *(repo, rebase) {
 });
 
 /**
- * @class {SubmoduleRebaser}
- * This is a mechanism class that maintains the state of a submodule through a
- * rebase operation.
- *
- * @constructor
- * Create a new `SubmoduleRebaser` for the specified `submodule` having the
- * specified `repo`.  Record commits (from new to old) in the specified
- * `submoduleCommits` map.
+ * Stage all entries in the specified `index` and return true, or return false
+ * if there is nothing to stage in `index`.
  *
- * @param {String}             submoduleName
- * @param {NodeGit.Repository} repo
- * @param {Object}             submoduleCommits writable
- * @param {SubmoduleFetcher}   fetcher
+ * @param {NodeGit.Index} index
+ * @return {Boolean}
  */
-function SubmoduleRebaser(submoduleName, repo, submoduleCommits, getFetcher) {
-    assert.isString(submoduleName);
-    assert.instanceOf(repo, NodeGit.Repository);
-    assert.isObject(submoduleCommits);
-    assert.isFunction(getFetcher);
+const stageAll = co.wrap(function *(index) {
```

Comment:
You might think so, but `updateAll` doesn't update "all", it updates all paths that match an array of wildcards.